docs/extended-tour.md

```diff
@@ -0,0 +1,159 @@
+# Extended Tour
+Welcome to the extended tour of node-cpp-skel. This documentation is especially handy if you're just starting out with building [Node Addon-ons](https://github.com/mapbox/cpp/blob/master/node-cpp.md) or writing C++ :tada:
+
+## Table of Contents:
+- [Walkthrough of example code](extended-tour.md#walkthrough-example-code)
+- [What does "build" mean?](extended-tour.md#what-does-build-mean)
+- [Software build components](extended-tour.md#software-build-components)
+- [Configuration files](extended-tour.md#configuration-files)
+- [Autogenerated files](extended-tour.md#autogenerated-files)
+- [Overall flow](extended-tour.md#overall-flow)
+
+# Walkthrough example code
+
+This skeleton includes a few examples of how you might design your application, including standalone functions and creating Objects/Classes. Both the synchronous and asynchronous versions of each are included to give an iterative example of what the progression from sync to async looks like. Let's run through reasons why you'd design your code in these ways:
+1. [Standalone function](API.md#hello-1)
+2. [Standalone asynchronous function](../API.md#hello_async)
+3. [Object/Class](../API.md#helloobject)
+4. [Asynchronous Object/Class](../API.md#helloobjectasync)
+
+
+### When would you use a standalone function?
+Standalone functions are simple and useful when your code is doing some kind of basic operation. They avoid the need for a class, since creating a class in this situation would add unnecessary complexity. Creating an asynchronous standalone function can be powerful if your code has a specific, yet expensive operation to do and would benefit from the use of multiple threads at one time to do that expensive work.
```

Comment:
Thanks @springmeyer , I like using `wonderful.callme()` to demonstrate. Relevant to mention anything about performance benefits by using standalone functions instead of object? For example, I have in my notes `In vt-shaver, we implemented a standalone funciton which is more performant because you dont need to create individual object instances for each use`. Perhaps this over complicates the definition, but seems relevant to the purpose of skel: performance.